File: NumberGuessingGame.lua
math.randomseed(os.time())
local n = math.random(1,100)
local guesses = 0

print("Guess the random number between 1 and 100!")


repeat
    print("Make a guess: ")
    local guess = tonumber(io.read())

    if guess == n then
        print("You win!")
        print("You guessed: "..guesses.." times!")

        print("Press ENTER to continue . . .")
        io.read()

    elseif guess > n then
        print("Your guess was too high, try again!")
        guesses = guesses + 1

    elseif guess < n then
        print("Your guess was too low, try again!")
        guesses = guesses + 1

    end

until guess == n

File: NumberGuessingGame.r
num = floor(runif(1, min=1, max=100))
guess = 0
guessCount = 0


while(TRUE) {
    cat("Guess the number between 1 and 100: ")

    guess = scan(n=1)

    guessCount = guessCount + 1

    if(guess < num) {
        cat("Your guess is too low. Try again.\n")
        
    } else if (guess > num) {
        cat("Your guess is too high. Try again.\n")

    } else {
        break
    }
}


cat("You win!\n")
cat("You guessed", guessCount, "times!\n")

cat("Press ENTER to continue . . .")
scan()
